File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Syntax/WithEventsPropertyEventContainerSyntax.cs
using Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax;

namespace Microsoft.CodeAnalysis.VisualBasic.Syntax
{
	public sealed class WithEventsPropertyEventContainerSyntax : EventContainerSyntax
	{
		internal WithEventsEventContainerSyntax _withEventsContainer;

		internal IdentifierNameSyntax _property;

		public WithEventsEventContainerSyntax WithEventsContainer => GetRedAtZero(ref _withEventsContainer);

		public SyntaxToken DotToken => new SyntaxToken(this, ((Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.WithEventsPropertyEventContainerSyntax)base.Green)._dotToken, GetChildPosition(1), GetChildIndex(1));

		public IdentifierNameSyntax Property => GetRed(ref _property, 2);

		internal WithEventsPropertyEventContainerSyntax(GreenNode green, SyntaxNode parent, int startLocation)
			: base(green, parent, startLocation)
		{
		}

		internal WithEventsPropertyEventContainerSyntax(SyntaxKind kind, DiagnosticInfo[] errors, SyntaxAnnotation[] annotations, WithEventsEventContainerSyntax withEventsContainer, PunctuationSyntax dotToken, IdentifierNameSyntax property)
			: this(new Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.WithEventsPropertyEventContainerSyntax(kind, errors, annotations, (Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.WithEventsEventContainerSyntax)withEventsContainer.Green, dotToken, (Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.IdentifierNameSyntax)property.Green), null, 0)
		{
		}

		public WithEventsPropertyEventContainerSyntax WithWithEventsContainer(WithEventsEventContainerSyntax withEventsContainer)
		{
			return Update(withEventsContainer, DotToken, Property);
		}

		public WithEventsPropertyEventContainerSyntax WithDotToken(SyntaxToken dotToken)
		{
			return Update(WithEventsContainer, dotToken, Property);
		}

		public WithEventsPropertyEventContainerSyntax WithProperty(IdentifierNameSyntax property)
		{
			return Update(WithEventsContainer, DotToken, property);
		}

		internal override SyntaxNode GetCachedSlot(int i)
		{
			return i switch
			{
				0 => _withEventsContainer, 
				2 => _property, 
				_ => null, 
			};
		}

		internal override SyntaxNode GetNodeSlot(int i)
		{
			return i switch
			{
				0 => WithEventsContainer, 
				2 => Property, 
				_ => null, 
			};
		}

		public override TResult Accept<TResult>(VisualBasicSyntaxVisitor<TResult> visitor)
		{
			return visitor.VisitWithEventsPropertyEventContainer(this);
		}

		public override void Accept(VisualBasicSyntaxVisitor visitor)
		{
			visitor.VisitWithEventsPropertyEventContainer(this);
		}

		public WithEventsPropertyEventContainerSyntax Update(WithEventsEventContainerSyntax withEventsContainer, SyntaxToken dotToken, IdentifierNameSyntax property)
		{
			if (withEventsContainer != WithEventsContainer || dotToken != DotToken || property != Property)
			{
				WithEventsPropertyEventContainerSyntax withEventsPropertyEventContainerSyntax = SyntaxFactory.WithEventsPropertyEventContainer(withEventsContainer, dotToken, property);
				SyntaxAnnotation[] annotations = GetAnnotations();
				if (annotations != null && annotations.Length > 0)
				{
					return SyntaxNodeExtensions.WithAnnotations(withEventsPropertyEventContainerSyntax, annotations);
				}
				return withEventsPropertyEventContainerSyntax;
			}
			return this;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Syntax/WithEventsPropertyEventContainerSyntax.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic
Imports Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
Imports System

Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax
	Public NotInheritable Class WithEventsPropertyEventContainerSyntax
		Inherits Microsoft.CodeAnalysis.VisualBasic.Syntax.EventContainerSyntax
		Friend _withEventsContainer As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsEventContainerSyntax

		Friend _property As Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax

		Public ReadOnly Property DotToken As Microsoft.CodeAnalysis.SyntaxToken
			Get
				Return New Microsoft.CodeAnalysis.SyntaxToken(Me, DirectCast(MyBase.Green, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.WithEventsPropertyEventContainerSyntax)._dotToken, Me.GetChildPosition(1), MyBase.GetChildIndex(1))
			End Get
		End Property

		Public ReadOnly Property [Property] As Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax
			Get
				Return MyBase.GetRed(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax)(Me._property, 2)
			End Get
		End Property

		Public ReadOnly Property WithEventsContainer As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsEventContainerSyntax
			Get
				Return MyBase.GetRedAtZero(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsEventContainerSyntax)(Me._withEventsContainer)
			End Get
		End Property

		Friend Sub New(ByVal green As GreenNode, ByVal parent As SyntaxNode, ByVal startLocation As Integer)
			MyBase.New(green, parent, startLocation)
		End Sub

		Friend Sub New(ByVal kind As SyntaxKind, ByVal errors As DiagnosticInfo(), ByVal annotations As SyntaxAnnotation(), ByVal withEventsContainer As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsEventContainerSyntax, ByVal dotToken As PunctuationSyntax, ByVal [property] As Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax)
			MyClass.New(New Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.WithEventsPropertyEventContainerSyntax(kind, errors, annotations, DirectCast(withEventsContainer.Green, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.WithEventsEventContainerSyntax), dotToken, DirectCast([property].Green, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.IdentifierNameSyntax)), Nothing, 0)
		End Sub

		Public Overrides Function Accept(Of TResult)(ByVal visitor As VisualBasicSyntaxVisitor(Of TResult)) As TResult
			Return visitor.VisitWithEventsPropertyEventContainer(Me)
		End Function

		Public Overrides Sub Accept(ByVal visitor As Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxVisitor)
			visitor.VisitWithEventsPropertyEventContainer(Me)
		End Sub

		Friend Overrides Function GetCachedSlot(ByVal i As Integer) As Microsoft.CodeAnalysis.SyntaxNode
			Dim syntaxNode As Microsoft.CodeAnalysis.SyntaxNode
			Dim num As Integer = i
			If (num = 0) Then
				syntaxNode = Me._withEventsContainer
			ElseIf (num = 2) Then
				syntaxNode = Me._property
			Else
				syntaxNode = Nothing
			End If
			Return syntaxNode
		End Function

		Friend Overrides Function GetNodeSlot(ByVal i As Integer) As SyntaxNode
			Dim withEventsContainer As SyntaxNode
			Dim num As Integer = i
			If (num = 0) Then
				withEventsContainer = Me.WithEventsContainer
			ElseIf (num = 2) Then
				withEventsContainer = Me.[Property]
			Else
				withEventsContainer = Nothing
			End If
			Return withEventsContainer
		End Function

		Public Function Update(ByVal withEventsContainer As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsEventContainerSyntax, ByVal dotToken As Microsoft.CodeAnalysis.SyntaxToken, ByVal [property] As Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax) As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsPropertyEventContainerSyntax
			Dim withEventsPropertyEventContainerSyntax As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsPropertyEventContainerSyntax
			If (withEventsContainer <> Me.WithEventsContainer OrElse dotToken <> Me.DotToken OrElse [property] <> Me.[Property]) Then
				Dim withEventsPropertyEventContainerSyntax1 As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsPropertyEventContainerSyntax = Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory.WithEventsPropertyEventContainer(withEventsContainer, dotToken, [property])
				Dim annotations As SyntaxAnnotation() = MyBase.GetAnnotations()
				withEventsPropertyEventContainerSyntax = If(annotations Is Nothing OrElse CInt(annotations.Length) <= 0, withEventsPropertyEventContainerSyntax1, withEventsPropertyEventContainerSyntax1.WithAnnotations(annotations))
			Else
				withEventsPropertyEventContainerSyntax = Me
			End If
			Return withEventsPropertyEventContainerSyntax
		End Function

		Public Function WithDotToken(ByVal dotToken As Microsoft.CodeAnalysis.SyntaxToken) As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsPropertyEventContainerSyntax
			Return Me.Update(Me.WithEventsContainer, dotToken, Me.[Property])
		End Function

		Public Function WithProperty(ByVal [property] As Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax) As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsPropertyEventContainerSyntax
			Return Me.Update(Me.WithEventsContainer, Me.DotToken, [property])
		End Function

		Public Function WithWithEventsContainer(ByVal withEventsContainer As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsEventContainerSyntax) As Microsoft.CodeAnalysis.VisualBasic.Syntax.WithEventsPropertyEventContainerSyntax
			Return Me.Update(withEventsContainer, Me.DotToken, Me.[Property])
		End Function
	End Class
End Namespace